File: src/main/java/co/volight/terrario/mixin/InGameHudAccessor.java
package co.volight.terrario.mixin;

import net.minecraft.client.MinecraftClient;
import net.minecraft.client.font.TextRenderer;
import net.minecraft.client.gui.hud.InGameHud;
import net.minecraft.entity.player.PlayerEntity;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;
import org.spongepowered.asm.mixin.gen.Invoker;

import java.util.Random;

@Mixin(InGameHud.class)
public interface InGameHudAccessor {
    @Accessor
    Random getRandom();

    @Accessor
    MinecraftClient getClient();

    @Invoker
    PlayerEntity callGetCameraPlayer();

    @Invoker
    TextRenderer callGetTextRenderer();

    @Accessor
    int getRenderHealthValue();

    @Accessor
    int getScaledWidth();

    @Accessor
    int getScaledHeight();
}

File: src/main/kotlin/co/volight/terrario/client/gui/InGameHudAccessor.kt
@file:Environment(value = EnvType.CLIENT)

package co.volight.terrario.client.gui

import co.volight.terrario.mixin.client.gui.InGameHudAccessor
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.font.TextRenderer
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.player.PlayerEntity

val InGameHudAccessor.cameraPlayer: PlayerEntity get() = callGetCameraPlayer()

val InGameHudAccessor.textRenderer: TextRenderer get()  = callGetTextRenderer()

val InGameHudAccessor.riddenEntity: LivingEntity? get() = callGetRiddenEntity()

fun InGameHudAccessor.getHeartCount(entity: LivingEntity?) = callGetHeartCount(entity)

fun InGameHudAccessor.getHeartRows(heartCount: Int) = callGetHeartRows(heartCount)
